modules/health_quest/app/services/health_quest/patient_generated_data/questionnaire_response/factory.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module HealthQuest
+  module PatientGeneratedData
+    module QuestionnaireResponse
+      class Factory
```

Comment:
I like the Factory pattern, though it should be noted this implementation *does* collide conceptually with the existing use of "factories" in the sense of `spec/factories` -> I'd be interested in your thoughts on this overall?

Does `QuestionnaireResponse` need module-level capabilities or could there be a manufacture method on that class instead of having a dedicated factory class?

Reply:
Hi, the Factory class is intended to be used as a "service" level object which combines responsibilities that I feel don't belong in the controller from a design perspective. I feel it's different from the ideas behind `spec/factories`. The Factory object is not "set in stone" by any means. Currently, it makes testing smaller and cleaner and it's managed to make it easier to implement single responsibility and open-close principles across the code layer. 

I don't wish to make a call one way or the other just quite yet because I still have a couple more services(talking to PGD) I need to code and implement and I think keeping things loosely coupled as is might be a better call than settling for the wrong abstraction right at this moment. 

In other words, we will be creating a PR refactoring once all PGD endpoints are covered and we hope to get your input and approval on the final product.